src/objects.rs

```diff
@@ -120,12 +122,12 @@ reference_wrappers! {
 ///
 /// This can be freely implemented for anything with a way to get a position.
 pub trait HasPosition {
-    fn pos(&self) -> Position;
+    fn pos(&self) -> Result<Position, ConversionError>;
```

Comment:
What would you think of just not implementing `HasPosition` and other traits for `PowerCreep` and instead requiring first converting to an "InstantiatedPowerCreep" or similar before being able to call relevant methods?

I'd just suggest it because changing the traits like this requires everyone to deal with usually-non-existent error cases dealing with every other object.

The other disadvantage would be that this makes methods like `.get_range_to` panic when called with a power creep, something that "seems" valid. Ideally we should be able to differentiate between things which can be put as an argument to that and not at compile time, with traits or different types, rather than erroring out at runtime.

Thoughts?

Reply:
Agreed, this kind of change to these methods makes them a lot more awkward to use - my initial attempt at implementation used an enum to wrap an alive-or-dead-power-creep, but that ended up being a mess - this sounds better ;)

The comparison I'm thinking of is that we'd have something akin to a `std::rc::Weak` upgrade into a `std::rc::Rc` - taking the maybe-alive-power-creep reference object reference and upgrading it to get a we-know-it's-alive object with room object properties, `HasId`, `Attackable` etc.

Because this is only applicable to the results of accessing `game::power_creeps` (and `game::creeps` in the #246 case) case and not for objects found via `find()`/`look()`, my thinking is to have `PowerCreep` as the 'fully alive power creep' representation (which would be the find/look result, intuitively) and to add `PowerCreepInfo` or `GamePowerCreep` or somesuch which would be the version of the reference that can be `upgrade()`-ed to a full `PowerCreep` if it's spawned on the current shard.

I could use some guidance on how to get this implemented - I can't quite get my head around how to get that 'upgrade()' function set up to get the underlying js ref moved from `ShrodingersPowerCreep` to `PowerCreep` during the `upgrade()`, to wrap it in the 'real' object type.

Reply:
Worked it out and went with `AccountPowerCreep`, should be a much less breaking changeset now :)

Reply:
Awesome!

> The comparison I'm thinking of is that we'd have something akin to a std::rc::Weak upgrade into a std::rc::Rc - taking the maybe-alive-power-creep object reference and upgrading it to get a we-know-it's-alive object with room object properties, HasId, Attackable etc.

This sounds like a sane comparison to me. The new scheme looks good!

Reply:
> `ShrodingersPowerCreep`

🤣 

I think I like it better than `AccountPowerCreep` 😛